ci_framework/roles/use_install_yamls/library/create_env_file.py

```diff
@@ -0,0 +1,176 @@
+
+options:
+    makefile_path:
+        description: Full path of the install_yamls makefile.
+        required: true
+        type: str
+    install_yamls_var:
+        description:
+            - List of exported install_yamls variable
+        required: true
+        type: dict
+    envfile_name:
+        description: name of the environment variable filename
+        required: false
+        type: string
+    check_var:
+        description: Check the environment var exists in the makefile
+        required: false
+        type: bool
+    create_yaml:
+        description: Create Ansible Environment Yaml file
+        required: false
+        type: bool
+author:
+    - Chandan Kumar (@raukadah)
+'''
+
+EXAMPLES = r'''
+# Create env var file
+- name: Create envfile for
+  become: true
+  create_env_file:
+    makefile_path: <path to make file>
+    install_yamls_var:
+        microshift: 1
+'''
+
+RETURN = r'''
+sucess:
+    description: env file created successfully
+    type: bool
+    returned: always
+    sample: "True"
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+import os
+import yaml
+
+# Check install_yamls makefile exists
+def _check_makefile(makefile_path):
+    """
+    Check whether install_yamls makefile exists or not
+    """
+    if os.path.exists(makefile_path):
+        return True
+    else:
+        return False
+
+# Return install_yamls vars
+def _get_makefile_vars(makefile_path):
+    if _check_makefile(makefile_path):
+        with open(makefile_path) as f:
+            content = f.read().split('\n')
+    # Get a list of vars
+    vars_list = [data.split('?=') for data in content if '?=' in data]
+    vars = {k.strip().lower(): v.strip().lower() for k,v in vars_list}
+    return vars        
+
+# Check if vars exists in install_yamls or not
+def _check_vars(makefile_path, users_var):
+    var_not_found = []
+    makefile_vars = _get_makefile_vars(makefile_path)
+    for key in users_var.keys():
+        if key not in makefile_vars.keys():
+            var_not_found.append(key)
+
+    return var_not_found
+
+# Create environment file
+def _create_file(envfile_name, users_var, output_dir):
+    env_path = os.path.join(output_dir, envfile_name)
+    with open(env_path, 'w') as f:
+        for k,v in users_var.items():
+            f.write(f"{k.upper()}={v}\n")
+    # Check whether file exists or not
+    if os.path.exists(env_path):
+        return True
+    else:
+        False
+
+# Create yaml file
+def _create_yaml_file(envfile, users_var, output_dir):
+    yaml_path = os.path.join(output_dir, envfile_name, '.yaml')
+    users_var_dict = {k.upper():v for k,v in users_var.items()}
+    with open(yaml_path, 'w') as file:
+        yaml.dump(users_var_dict,file)    
+
+# Run the module
+def create_env_file(envfile, makefile, check_var, users_var, output_dir, create_yaml=False):
+    unwanted_vars = []
+    if check_var:
+        unwanted_vars = _check_vars(makefile, users_var)
```

Comment:
So, if there is unwanted_vars, the module will fail with None and no meaning message like: "variable x is not valid"